pkg/environment/api/project.go

```diff
@@ -284,6 +295,167 @@ func (s *EnvironmentService) CreateProject(
 	return &environmentproto.CreateProjectResponse{Project: project.Project}, nil
 }
 
+func (s *EnvironmentService) createProjectNoCommand(
+	ctx context.Context,
+	req *environmentproto.CreateProjectRequest,
+	localizer locale.Localizer,
+	editor *eventproto.Editor,
+) (*environmentproto.CreateProjectResponse, error) {
+	if err := validateCreateProjectRequestNoCommand(req, localizer); err != nil {
+		return nil, err
+	}
+	newProj, err := domain.NewProject(
+		strings.TrimSpace(req.Name),
+		strings.TrimSpace(req.UrlCode),
+		req.Description,
+		editor.Email,
+		req.OrganizationId,
+		false,
+	)
+	if err != nil {
+		return nil, err
+	}
+	var domainEvent *evdomain.Event
+	err = s.mysqlClient.RunInTransactionV2(ctx, func(ctxWithTx context.Context, tx mysql.Transaction) error {
+		storage := v2es.NewProjectStorage(tx)
+		domainEvent, err = s.newCreateDomainEvent(
+			ctx,
+			newProj.Project,
+			&environmentproto.Project{},
+			req,
+			editor,
+			localizer,
+		)
+		if err != nil {
+			return err
+		}
+		return storage.CreateProject(ctxWithTx, newProj)
+	})
+	if err != nil {
+		return nil, s.reportCreateProjectRequestError(ctx, req, err, localizer)
+	}
+	if err := s.publisher.Publish(ctx, domainEvent); err != nil {
+		return nil, s.reportCreateProjectRequestError(ctx, req, err, localizer)
+	}
```

Comment:
Moved the publish domain event from the MySQL transaction for creating and updating APIs.